### DISCUSSION TITLE
Asserting with or condition?

### DISCUSSION THREAD

**Discussion by m1212e** [cypress-io/cypress#17197] [Questions and Help] [answered]
Hi,

so I know it is possible to assert with an AND condition like this:
`cy.get('[data-test=selector]').should('not.exist').and('not.be.visible');`

But in my case, I just want to check if an error message is visible. I'm not interested in how it's done. (Doesn't matter if `display: none` or removed from the DOM via. if condition with the framework)
So is it possible to assert with an OR condition like this:
`cy.get('[data-test=selector]').should('not.exist').or('not.be.visible');` ?
And if not: Why and how would you do it instead?
Thanks for your time.

**Comment by jennifer-shehane** [comment]
I can't think of a way to do this with the way that Cypress will automatically assert on `cy.get()` that the element exists. I can see the usefulness of this though, but we've erred on the side of specificity. There were some false positives possible before when Cypress used to loosely enforce `.not.be.visible` to mean it's not in the DOM or not visible.

**Reply by m1212e** [reply]
So what you're basically saying is that it can't be done and for the reasons you provided in your answer also never will be possible to do?
I see your point and obviousy I agree, but for this specific case it would be pretty nice to have.

**Reply by jennifer-shehane** [reply]
We don't have any current plans to support this. Just explaining past decisions on why this assertion was made more specific.

**Comment by bahmutov** [comment, marked as answer]
You can always write your own assertion logic, the simplest would be to write .should(callback) function

**Reply by m1212e** [reply]
Probably the best solution, thanks!
